src/Libraries/CoreNodeModelsWpf/CurveMapper/CurveMapperControl.xaml.cs

```diff
@@ -463,4 +467,102 @@ private void ThumbResizeThumbOnDragDeltaHandler(object sender, DragDeltaEventArg
             curveMapperNodeModel.GenerateRenderValues();
         }
     }
+
+    /// <summary>
+    /// Converts double values to compact strings with optional scientific notation and superscripts.
+    /// </summary>
+    public class NumberFormatterConverter : IValueConverter
```

Comment:
I would suggest moving this to the converters class under DynamoCoreWpf. Can be used for other UI nodes as well.